tornado/test/testing_test.py

```diff
@@ -75,6 +76,28 @@ def fail(message):
         self.assertEqual(str(cm.exception), "error one")
 
 
+class AsyncHTTPTestCaseTest(AsyncHTTPTestCase):
+    def get_app(self):
+        return Application()
+
+    def test_fetch_segment(self):
+        path = '/path'
+        response = self.fetch(path)
+        self.assertEqual(response.request.url, self.get_url(path))
+
+    def test_fetch_full_http_url(self):
+        path = 'http://localhost/path'
```

Comment:
It's kind of rude to generate 404s (probably) every time you run the tests if you happen to be running a local web server on port 80. It's better to either use a port that we know won't work (you can use `bind_unused_port()` to allocate one), or use the HTTP server created by AsyncHTTPTestCase and verify that the url doesn't get mangled along the way (e.g. by creating a handler that does `self.write(self.request.path)`